lib/Request.js

```diff
@@ -38,6 +38,27 @@ class Request {
   }
 
   /**
+   * Rename the key (if rename function exists) when used in map or filter.
+   *
+   * @param {String} key The string to rename
+   * @return {String} key The resulting string
+   */
+
+  rename(key) {
+    // Rename if rename function exists
+    if (this.target.rename) {
```

Comment:
since the name of this method is `rename`, this logic should be in the map/filter functions (since sometimes it won't rename)